app/src/ext_power_generic.c

```diff
@@ -167,6 +174,39 @@ static int ext_power_generic_init(const struct device *dev) {
     return 0;
 }
 
+#ifdef CONFIG_DEVICE_POWER_MANAGEMENT
+static int ext_power_generic_pm_control(const struct device *dev, uint32_t ctrl_command,
+                                        void *context, device_pm_cb cb, void *arg) {
+    int rc;
+    struct ext_power_generic_data *data = dev->driver_data;
+
+    switch (ctrl_command) {
+    case DEVICE_PM_SET_POWER_STATE:
+        if (*((uint32_t *)context) == DEVICE_PM_ACTIVE_STATE) {
+            data->pm_state = DEVICE_PM_ACTIVE_STATE;
```

Comment:
Shouldn't this also be calling `ext_power_generic_enable`?

Reply:
Not in this case. When the device turns back on from sleep, the ext_power device is restarted and will automatically turn ext power back to its last saved state from settings. When we disable before going to sleep, that off state isn't saved in settings, so we'll properly retrieve the last state.

If we explicitly enabled here, people's power cut offs would turn back on whenever they wake up and get saved that way.

Reply:
My point is, when we add per-device power state management, we will want to cut/restore power state based on idle/active.

So we'll need an API for "on/off" that isn't state persisting. So we can call *that* from here.

Doesn't need to be this PR, so I'm going to approve for now.

We also need to consider that other devices may say "no, don't sleep now!" and abort the sleep state change, I believe.